front/lib/resources/membership_resource.ts

```diff
@@ -0,0 +1,344 @@
+import type {
+  MembershipRoleType,
+  RequireAtLeastOne,
+  Result,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type {
+  Attributes,
+  InferAttributes,
+  ModelStatic,
+  Transaction,
+  WhereOptions,
+} from "sequelize";
+import { Op } from "sequelize";
+
+import type { Workspace } from "@app/lib/models";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import { MembershipModel } from "@app/lib/resources/storage/models/membership";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+import logger from "@app/logger/logger";
+
+type GetMembershipsOptions = RequireAtLeastOne<{
+  userIds: number[];
+  workspace: WorkspaceType | Workspace;
+}> & {
+  roles?: MembershipRoleType[];
+};
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface MembershipResource
+  extends ReadonlyAttributesType<MembershipModel> {}
+export class MembershipResource extends BaseResource<MembershipModel> {
+  static model: ModelStatic<MembershipModel> = MembershipModel;
+
+  constructor(
+    model: ModelStatic<MembershipModel>,
+    blob: Attributes<MembershipModel>
+  ) {
+    super(MembershipModel, blob);
+  }
+
+  static async getActiveMemberships({
+    userIds,
+    workspace,
+    roles,
+  }: GetMembershipsOptions): Promise<MembershipResource[]> {
+    const whereClause: WhereOptions<InferAttributes<MembershipModel>> = {
+      startAt: {
+        [Op.lte]: new Date(),
+      },
+      endAt: {
+        [Op.or]: [{ [Op.eq]: null }, { [Op.gte]: new Date() }],
+      },
+    };
+
+    if (userIds) {
+      whereClause.userId = userIds;
+    }
+    if (workspace) {
+      whereClause.workspaceId = workspace.id;
+    }
+    if (roles) {
+      whereClause.role = {
+        [Op.in]: roles,
+      };
+    }
+
+    const memberships = await MembershipModel.findAll({
+      where: whereClause,
+    });
+
+    return memberships.map(
+      (membership) => new MembershipResource(MembershipModel, membership.get())
+    );
+  }
+
+  static async getLatestMemberships({
+    userIds,
+    workspace,
+    roles,
+  }: GetMembershipsOptions): Promise<MembershipResource[]> {
+    const orderedResourcesFromModels = (resources: MembershipModel[]) =>
+      resources
+        .sort((a, b) => a.startAt.getTime() - b.startAt.getTime())
+        .map(
+          (resource) => new MembershipResource(MembershipModel, resource.get())
+        );
+
+    const where: WhereOptions<InferAttributes<MembershipModel>> = {};
+    if (roles) {
```

Comment:
I believe it might, but I am not convinced that inline ternaries are bringing a lot of extra readability here.

Reply:
Ok I understand; 
I advocate for it because it turns 13 lines into 3, and while I agree that ternaries are hard to read in some cases e.g. when nested, flat ternaries are the equivalent of saying "hi" vs "good morning"; legit part of the language (that we commonly use) and we should learn to read them as fluently as as we learned to read the `if`s

Clearly a nit, will approve anyway

Reply:
Alright you flipped me :) changing